gnupg: 2.1.14 fails to install on Darwin (ld: symbol(s) not found for architecture x86_64)

## Issue description

gnupg 2.1.14 fails to build on Darwin. 

### Steps to reproduce

```
$ nix-env -i gnupg
...
Undefined symbols for architecture x86_64:
  "_libintl_bind_textdomain_codeset", referenced from:
      _i18n_switchto_utf8 in libcommon.a(libcommon_a-i18n.o)
      _i18n_switchback in libcommon.a(libcommon_a-i18n.o)
      _i18n_utf8 in libcommon.a(libcommon_a-i18n.o)
  "_libintl_bindtextdomain", referenced from:
      _i18n_init in libcommon.a(libcommon_a-i18n.o)
      _i18n_localegettext in libcommon.a(libcommon_a-i18n.o)
  "_libintl_gettext", referenced from:
      _my_strusage in gpgscm-main.o
      _create_pipe_and_estream in libcommon.a(libcommon_a-exechelp-posix.o)
      _gnupg_spawn_process in libcommon.a(libcommon_a-exechelp-posix.o)
      _gnupg_spawn_process_fd in libcommon.a(libcommon_a-exechelp-posix.o)
      _gnupg_wait_processes in libcommon.a(libcommon_a-exechelp-posix.o)
      _gnupg_spawn_process_detached in libcommon.a(libcommon_a-exechelp-posix.o)
      _disable_core_dumps in libcommon.a(libcommon_a-sysutils.o)
      ...
  "_libintl_setlocale", referenced from:
      _i18n_init in libcommon.a(libcommon_a-i18n.o)
      _i18n_localegettext in libcommon.a(libcommon_a-i18n.o)
  "_libintl_textdomain", referenced from:
      _i18n_init in libcommon.a(libcommon_a-i18n.o)
ld: symbol(s) not found for architecture x86_64
```

## Technical details

* System: Mac OS X 10.11.5
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: "16.09pre88082.f242b75"

I have also tried the nixpkgs master branch at:

* commit 9626707e2b796c04871a22a583500a45c1c436d7
* Date:   Tue Aug 9 19:21:58 2016 +0200

--- a/pkgs/tools/security/gnupg/21.nix
+++ b/pkgs/tools/security/gnupg/21.nix
@@ -27,6 +27,8 @@ stdenv.mkDerivation rec {
     readline libusb gnutls adns openldap zlib bzip2
   ];
 
+  patches = [ ./fix-gpgsm-linking.patch ];
+
   postPatch = stdenv.lib.optionalString stdenv.isLinux ''
     sed -i 's,"libpcsclite\.so[^"]*","${pcsclite}/lib/libpcsclite.so",g' scd/scdaemon.c
   ''; #" fix Emacs syntax highlighting :-(

--- a/pkgs/tools/security/gnupg/21.nix
+++ b/pkgs/tools/security/gnupg/21.nix
@@ -27,6 +27,8 @@ stdenv.mkDerivation rec {
     readline libusb gnutls adns openldap zlib bzip2
   ];
 
+  # gpgsm-linking is fixed by commit (c49c43d7) in the gnupg master branch;
+  # fix-gpgsm-linking.patch should be dropped after gnupg 2.1.15 is released
   patches = [ ./fix-gpgsm-linking.patch ];
 
   postPatch = stdenv.lib.optionalString stdenv.isLinux ''

--- a/pkgs/tools/security/gnupg/fix-gpgsm-linking.patch
+++ b/pkgs/tools/security/gnupg/fix-gpgsm-linking.patch
@@ -1,0 +1,11 @@
+--- a/tests/gpgscm/Makefile.in
++++ b/tests/gpgscm/Makefile.in
+@@ -457,7 +457,7 @@
+ 	scheme-config.h opdefines.h scheme.c scheme.h scheme-private.h
+ 
+ gpgscm_LDADD = $(LDADD) $(common_libs) \
+-	$(NETLIBS) $(LIBICONV) $(LIBREADLINE) \
++	$(NETLIBS) $(LIBICONV) $(LIBREADLINE) $(LIBINTL) \
+ 	$(LIBGCRYPT_LIBS) $(GPG_ERROR_LIBS)
+ 
+ t_child_SOURCES = t-child.c

